test/test_queue.js

```diff
@@ -1411,7 +1411,11 @@ describe('Queue', function() {
         expect(job.data.foo).to.be.eql('bar');
         expect(err).to.be.eql(notEvenErr);
         failedOnce = true;
-        job.retry();
+        job.retry().then(function() {
+          expect(job.failedReason).to.be.null;
+          expect(job.processedOn).to.be.null;
+          expect(job.finishedOn).to.be.null;
```

Comment:
would be great to actually do a ```Queue.getJob()``` and check that they are also null after reading the job from redis.

Reply:
After the retry() promise resolves (inside `then`) you mean? Because, I do not think they will be null after the promise resolves (?)

Reply:
I updated the test, let me know if you meant something else.

Reply:
yes, it should only be null the failedReason since now it has been completed.
This makes me think however that the user would loose the information on why a job failed before the retry, what do you think?

Reply:
They can always check failedReason before they execute `retry`. They have the opportunity to check it - if they want - before doing anything else so I think that the information is not lost since nothing automatically happens.

How about the other two fields though? Why aren't they timestamp integer after the retry completes?

Reply:
it is only returning the job as it is in that very moment

Reply:
Ok got it now, thanks for clarifying :)